scene/gui/color_picker.cpp

```diff
@@ -1015,20 +1215,59 @@ void ColorPicker::_w_input(const Ref<InputEvent> &p_event) {
 	}
 }
 
-void ColorPicker::_preset_input(const Ref<InputEvent> &p_event, const Color &p_color) {
+void ColorPicker::_slider_or_spin_input(const Ref<InputEvent> &p_event) {
+	if (line_edit_mouse_release) {
+		line_edit_mouse_release = false;
+		return;
+	}
+	Ref<InputEventMouseButton> bev = p_event;
+	if (bev.is_valid() && !bev->is_pressed() && bev->get_button_index() == MouseButton::LEFT) {
+		_add_recent_preset_button(true);
+	}
+}
+
+void ColorPicker::_line_edit_input(const Ref<InputEvent> &p_event) {
+	Ref<InputEventMouseButton> bev = p_event;
+	if (bev.is_valid() && !bev->is_pressed() && bev->get_button_index() == MouseButton::LEFT) {
+		line_edit_mouse_release = true;
```

Comment:
What is this for?

Reply:
We want to add a recent preset when mouse is released after using spinner of SpinBox but it gets applied for it's LineEdit too, so we need an early return if there's a mouse release in LineEdit. Don't know if it's the right way to do this.

Reply:
Eh, SpinBox makes my cursor disappear so I can't test it :/
Maybe `value_changed` signal would be enough.

Reply:
No, as `color_changed` gets emitted for every mouse motion event but we only want to add a preset on mouse release, `deferred_mode` only works for `uv_edit`, `w_edit` and `wheel_edit`.